### PR TITLE
Fix url splitting

### PR DESCRIPTION
**Pull request by MartinquaXD**
# Description
Fixes https://github.com/cowprotocol/services/issues/1988

# Changes
Added a helper function to split a URL into base and path which is used for the `legacy` solver adapter.

## How to test
added a unit test

### CODE REVIEW

**Review by sunce86 — APPROVED** [review]
Lg

**Comment by MartinquaXD** [comment]
Merging early since this is needed for enabling API keys for one of the solvers and makes managing our current infra config easier.
Feel free to review anyway. I will then address suggestions afterwards.

File: crates/shared/src/url.rs [hunk 1]
```diff
@@ -12,3 +15,66 @@ pub fn join(url: &Url, mut path: &str) -> Url {
...
 }
+
+/// Splits a URL right where the path begins into base and endpoint.
+/// https://my.solver.xyz/solve/1?param=1#fragment=some
+/// becomes
+/// (https://my.solver.xyz/, /solve/1?param=1#fragment=some)
+/// Path that were split like this can be joined to the original URL using
+/// [`join`].
+pub fn split_at_path(url: &Url) -> Result<(Url, String)> {
+    let base = format!(
```
Comment (fleupold) [inline crates/shared/src/url.rs:26]: nit: If I'm not mistaken, this should be the same as `url.origin().unicode_serialization`

https://docs.rs/url/latest/url/struct.Url.html#method.origin

**Review by fleupold — COMMENTED** [review]
It's not entirely clear to me why we even differentiate between base and solve_path in the HTTP API constructor? It looks like we reuse base to append "/notify". Probably better to have this logic there?

But this works as well.

**Comment by MartinquaXD** [comment]
> It's not entirely clear to me why we even differentiate between base and solve_path in the HTTP API constructor?

The idea was to have the legacy wrapper just be a wrapper and not require changes for a component we are phasing out anyway.

File: crates/shared/src/url.rs (hunk 1 above)
Comment (MartinquaXD) [reply crates/shared/src/url.rs:26]: Was looking for something like this but couldn't find it. 👍

**Pull request merged**